src/dimensions.js

```diff
@@ -18,16 +18,28 @@ export function Dimensions(
     margin,
     marginTop = margin !== undefined
       ? margin
-      : Math.max((xAxis === "top" ? 30 : 0) + facetMarginTop, yAxis || fyAxis ? 20 : 0.5 - offset),
+      : Math.max(
+          (projection == null && xAxis === "top" ? 30 : 0) + facetMarginTop,
```

Comment:
I think rather than checking `projection == null` here, we should ensure that xAxis is null if the projection is not null. I’ll look into this…

Reply:
Actually, that should already be the case, since in inferScaleType we prevent the _x_ and _y_ scale from being instantiated if there is a projection, and if there is no _x_ or _y_ scale, there is no corresponding axis. So I don’t think this code is doing anything and can be safely removed (and the tests pass, so…).